fe/fe-core/src/main/java/org/apache/doris/qe/StreamLoadTxnExecutor.java

```diff
@@ -42,20 +42,31 @@
 
 import org.apache.thrift.TException;
 
+import java.nio.ByteBuffer;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
-public class InsertStreamTxnExecutor {
+public class StreamLoadTxnExecutor {
     private long txnId;
     private TUniqueId loadId;
-    private TransactionEntry txnEntry;
+    private final TransactionEntry txnEntry;
+    private final TFileFormatType formatType;
 
-    public InsertStreamTxnExecutor(TransactionEntry txnEntry) {
-        this.txnEntry = txnEntry;
+    private final TFileCompressType compressType;
+
+    public StreamLoadTxnExecutor(TransactionEntry txnEntry) {
+        this(txnEntry, TFileFormatType.FORMAT_PROTO, TFileCompressType.PLAIN);
+    }
+
+    public StreamLoadTxnExecutor(TransactionEntry entry, TFileFormatType formatType, TFileCompressType compressType) {
+        this.txnEntry = entry;
+        this.formatType = formatType;
+        this.compressType = compressType;
     }
 
     public void beginTransaction(TStreamLoadPutRequest request) throws UserException, TException, TimeoutException,
```

Comment:
How about name it `executeTranscation`?

Reply:
In the `Transcation insert`, it have 3 step: begin/insert/commit for user.
But the for the be, the begin txn is called before the this `InsertStreamTxnExecutor`